CONTRIBUTING.md

```diff
@@ -0,0 +1,62 @@
+| `.gitignore`      | A list of files and folders which won't be tracked by Git.                                                                                                                                                                          |
+| `.eslintrc.json`  | [ESLint configuration data](https://eslint.org/docs/user-guide/configuring).                                                                                                                                                        |
+| `.eslintignore`   | A list of files and folders which won't be passed through ESLint                                                                                                                                                                    |
+| `AUTHORS.md`      | List of copyright holders and contributors                                                                                                                                                                                          |
+| `Gruntfile.js`    | Grunt file used to copy data from node_modules and do linting process.                                                                                                                                                              |
+| `package.json`    | Contains scripts and a list of dependencies required to build the site, as well as general information about the repository                                                                                                                                                                             |
+| `README.md`       | Read Me file provides introduction to the repository.                                                                                                                                                                               |
+
+**Contributions to the project can be made in following ways:**
+
+1. *Contribution towards building features for the project.*
+
+* There are several features that still require attention.
+  * Improving the User interface of submitted answers.
+  * Search functionality among the site data.
+  * Support for Comments.
+
+1. *Submitting new blog posts or news articles*
+
+* Post submission through Github:
+  * Go to the `src/news` folder
+  * Create a new Markdown file with title in the following format - `YYYY-MM-DD-titleOfNewsArticle.md`.
+  * Use the YYYY-MM-DD-newsarticlename.md file as template.
+  * Update the date, title and permalink in the front matter.
+  * Your article must have a specific template as mentioned below to get merged. make sure to follow the
+    template while submitting the answer.
+  * The template is as follows:
+   `---`
+   `layout: layouts/post`
+   `title: Title of News Article`
+   `date: 'YYYY-MM-DD'`
+   `tags: post`
+   `permalink: news/YYYY-MM-DD-newsArticleName.html`
+   `---`
+   `Content Data`
```

Comment:
This documentation get updated including documenting how output filenames are generated from the `date`, and `title` or `filename` value

Also should document that `filename` should be used in situations where the title may be too long and a shorter filename is desirable. It is also used to ensure consistency for legacy news posts.

Reply:
fixed